cocos/renderer/gfx-metal/MTLDevice.mm

```diff
@@ -347,5 +354,11 @@
     static_cast<CCMTLCommandBuffer *>(_cmdBuff)->copyBuffersToTexture(buffers, texture, regions, count);
 }
 
+void CCMTLDevice::onMemoryWarning() {
+    for (int i = 0; i < MAX_FRAMES_IN_FLIGHT; ++i) {
+        _gpuStagingBufferPools[i]->shrinkSize();
+    }
```

Comment:
How do you know which staging buffer pool can be shrink?

Reply:
it's kind of "try-to-shrink", if all the buffers in the buffer pool are in use, shrinkSize does nothing. Only when pool size greater than 1 and there are buffers can be released(curOffet == 0), shrinkSize release and erase them from buffer pool.

Reply:
Sounds good.